src/main.rs

```diff
@@ -419,4 +467,10 @@ mod tests {
 
         assert_eq!(ips.is_empty(), true);
     }
+    #[test]
+    fn parse_hosts_file_and_incorrect_hosts() {
```

Comment:
We are missing an assertion here ( `assert_eq` ) with the ips that were parsed correctly. 😄

Reply:
Good catch! Fixed.

Reply:
Just for future reference, the current way of testing domains might fail.

Our DNS resolver can return multiple IPs for 1 domain name, and we cannot guarantee that Google or example.com will return 1 IP each time.

Reply:
Good point 😄 I will revisit the way we test at some point, ideally our tests shouldn't fail when an external party fails.